test/e2e/helpers.go

```diff
@@ -823,3 +829,68 @@ func getPodLogs(ctx context.Context, clientset *kubernetes.Clientset, pod corev1
 	}
 	return b.String()
 }
+
+func InstallHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, repoURL, chartName, releaseName string, values []string) {
+	// Ensure the workload cluster kubeconfig secret exists before getting the workload cluster clusterProxy object
+	Eventually(func() error {
+		cl := input.ClusterProxy.GetClient()
+		secret := &corev1.Secret{}
+		key := client.ObjectKey{
+			Name:      fmt.Sprintf("%s-kubeconfig", input.ConfigCluster.ClusterName),
+			Namespace: input.ConfigCluster.Namespace,
+		}
+		err := cl.Get(ctx, key, secret)
+		if err != nil {
+			return err
+		}
+		return nil
+	}, input.WaitForControlPlaneIntervals...).Should(Succeed())
+	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
```

Comment:
Unfortunately in my testing I encountered an edge case where `input.ClusterProxy.GetWorkloadCluster` can produce a 404 grabbing the secret (and it's not retryable as there are embedded ginkgo expects in that func), so I put the above `Eventually` wrapper to ensure that we register a successful kubeconfig secret retrieval before proceeding.